eZ/Publish/API/Repository/Tests/ObjectStateServiceTest.php

```diff
@@ -376,6 +382,55 @@ public function testLoadObjectStateGroupThrowsNotFoundException()
         /* END: Use Case */
     }
 
+    /**
+     * @covers \eZ\Publish\API\Repository\ObjectStateService::loadGroupByIdentifier
+     */
+    public function testLoadObjectStateGroupByIdentifier(): void
+    {
+        $repository = $this->getRepository();
+
+        // $objectStateGroupIdentifier contains the identifier of the standard object state
+        // group ez_lock.
+        $objectStateService = $repository->getObjectStateService();
+
+        $loadedObjectStateGroup = $objectStateService->loadObjectStateGroupByIdentifier(
+            self::EXISTING_OBJECT_STATE_GROUP_IDENTIFIER
+        );
+
+        $this->assertInstanceOf(
+            ObjectStateGroup::class,
+            $loadedObjectStateGroup
+        );
+
+        $this->assertPropertiesCorrect(
+            [
+                'id' => 2,
+                'identifier' => 'ez_lock',
+                'mainLanguageCode' => 'eng-US',
+                'languageCodes' => ['eng-US'],
+                'names' => ['eng-US' => 'Lock'],
+                'descriptions' => ['eng-US' => ''],
+            ],
+            $loadedObjectStateGroup
+        );
+    }
+
+    /**
+     * @covers \eZ\Publish\API\Repository\ObjectStateService::loadGroupByIdentifier
+     */
+    public function testLoadObjectStateGroupByIdentifierThrowsNotFoundException(): void
+    {
+        $this->expectException(NotFoundException::class);
+
+        $repository = $this->getRepository();
+
+        $objectStateService = $repository->getObjectStateService();
+        // Throws a not found exception
```

Comment:
I would move `$this->expectException(NotFoundException::class);` here. Just before place when we expect exception

Reply:
It would be great to check a message